File: app/src/main/java/piotr/michalkiewicz/mealplannerclient/view/login_and_signup/LoginLoadingDialog.java
package piotr.michalkiewicz.mealplannerclient.view.login_and_signup;

import android.app.Activity;
import android.app.AlertDialog;
import android.view.LayoutInflater;

import com.mealplannerclient.R;

public class LoginLoadingDialog {

    private Activity activity;
    private AlertDialog alertDialog;

    public LoginLoadingDialog(Activity activity){
        this.activity = activity;
    }

    public void startLoadingDialog(){
        AlertDialog.Builder builder = new AlertDialog.Builder(activity);

        LayoutInflater inflater = activity.getLayoutInflater();
        builder.setView(inflater.inflate(R.layout.loading_dialog,null));
        builder.setCancelable(false);

        alertDialog = builder.create();
        alertDialog.show();
    }

    public void dismissDialog(){
        alertDialog.dismiss();
    }

}

File: app/src/main/java/piotr/michalkiewicz/mealplannerclient/view/login/LoginLoadingDialog.kt
package piotr.michalkiewicz.mealplannerclient.view.login

import android.app.AlertDialog
import android.content.Context
import android.view.LayoutInflater
import piotr.michalkiewicz.mealplannerclient.R
import piotr.michalkiewicz.mealplannerclient.view.MainActivity

class LoginLoadingDialog {

    private lateinit var alertDialog: AlertDialog
    private val context = MainActivity.getMainContext()

    fun startLoadingDialog() {
        val builder = AlertDialog.Builder(context)
        val inflater: LayoutInflater =
            context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        builder.setView(inflater.inflate(R.layout.loading_dialog, null))
        builder.setCancelable(false)

        alertDialog = builder.create()
        alertDialog.show()
    }

    fun dismissDialog() {
        alertDialog.dismiss()
    }
}